### PR TITLE
Project: Bare Metal Forms and Helpers

### PR DESCRIPTION
**Pull request by edxco**
### In this project, we

- Create User model
- Create three forms(palin html, form_tag and a form_for) in the new.html.erb and in edit.html.erb
- Add in user_controller the action new, create, edit, update

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# frozen_string_literal: true
+
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  root to: 'users#new'
+  resources :users, only: %i[new create edit update show destroy]
```
Comment (peterrobert) [inline config/routes.rb]: Your routes should be restricted to only: `resources :users, only: [:new, :create, :edit, :update]`
routes

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+  # def create
+  #  @user = User.new(username: params[:username], email: params[:email], password: params[:password])
+  #  if @user.save
+  #    redirect_to new_user_path
+  #  else
+  #    render :new
+  #  end
+  # end
+
```
Comment (peterrobert) [inline app/controllers/users_controller.rb]: Remove all the commented code.

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    @user = User.new(user_params)
+
+    if @user.save
+      flash.notice = "User '#{@user.username}' Created!"
+      redirect_to user_path(@user)
+    else
+      flash.notice = "Error, creating '#{@user.username}'!"
+      render :new
+    end
+  end
```
Comment (peterrobert) [inline app/controllers/users_controller.rb]: If the user is saved, 
redirect_to edit_user_path

**Review by peterrobert — CHANGES_REQUESTED** [review]
###  CHANGES REQUIRED :recycle:  :stop_sign: 
HELLO @edxco  AND @Blaise-Shyaka  :wave: VERY NICE PROGRESS SO FAR YOU HAVE DONE A GOOD JOB. HOWEVER, THERE ARE A FEW REQUIRED CHANGES TO BE MADE KINDLY CHECK BELOW :arrow_down: 

**IF YOU HAVE ANY QUESTION KINDLY REACH ME ON SLACK** @PETERROBERT

### HAPPY CODING :computer:

**Review by rOluochKe — APPROVED** [review]
Hello Team
### Well done :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next.
- Happy coding!

**Pull request merged**